src/parser.ts

```diff
@@ -153,9 +153,14 @@ export class Parser {
 
       if (tok.name === 'EOF') {
         break;
-      } else if (tok.name === 'parabreak') {
+      } else if (tok.name === 'parabreak' && this._t.peek(2).name !== 'EOF') {
```

Comment:
Hm. Rather than invoking the whole `parseText` infrastructure for the case where `tok.name === 'parabreak' && this._t.peek(2).name === 'EOF'`, could we not just handle that case explicitly here? as in

```js
} else if (tok.name === 'parabreak') {
  if (this._t.peek(2).name === 'EOF') {
    this.pushPos();
    this._t.next();
    pushOrJoin(frag, this.finish({ name: 'text', contents: tok.contents }));
  }
  break;
} else if (tok.name === 'text' || tok.name === 'whitespace' || tok.name === 'linebreak') {
```

Then the changes to `parseText` would be unnecessary. (Which I'd prefer; it strikes me as odd to have `parseText` stop when it hits a parabreak, but still consume the parabreak.)

Reply:
That doesn't pass the "trailing-long-parabreak" test in which there are more than two line feeds after the last "real" content ("**Error: expecting EOF, got parabreak**"). I think the stopping without consumption is weird as well, but it's preexisting and I'm not sure if changing that would break anything.

Reply:
I just pushed up a commit with the thing I was suggesting, and it seems to pass on CI. Feel free to revert if you think there's some other problem with it, though.

How were you observing the failure?